types/connect/index.d.ts

```diff
@@ -18,11 +18,15 @@ declare function createServer(): createServer.Server;
 declare namespace createServer {
     export type ServerHandle = HandleFunction | http.Server;
 
+    export class IncomingMessage extends http.IncomingMessage {
+        originalUrl?: http.IncomingMessage["url"];
```

Comment:
I'm not a TypeScript expert. What does this line do?

Reply:
`http.IncomingMessage["url"]` is a reference to the type of the `url` property in `http.IncomingMessage`, so it's setting the type of `originalUrl` to be the same as the type of `url`. It's called lookup types, [documented here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html#keyof-and-lookup-types).